Nama Mahasiswa    : Nicholaus Verdhy Putranto

Asal Universitas  : Universitas Dian Nuswwantoro Semarang

Kode Peserta      : PYTN - KS11 - 023

# Import Libray Yang dibutuhkan

>1. NumPy adalah library Python yang digunakan untuk melakukan komputasi numerik dalam bahasa Python.

>2. Pandas adalah library Python yang digunakan untuk melakukan analisis data.

>3. Seaborn adalah library Python yang digunakan untuk melakukan visualisasi data.

>4. Statsmodels adalah library Python yang digunakan untuk melakukan analisis statistik.

>5. Matplotlib adalah library Python yang digunakan untuk membuat visualisasi data secara umum.

>6. Math adalah library Python yang digunakan untuk melakukan operasi matematika dasar.

>7. Statistics adalah library Python yang digunakan untuk melakukan operasi statistik dasar.

>8. Scipy.stats adalah library Python yang digunakan untuk melakukan analisis statistik lanjutan.

>7. Warnings adalah library Python yang digunakan untuk menghasilkan peringatan saat ada situasi yang tidak diharapkan terjadi pada kode program.

In [47]:
import numpy as np
import pandas as pd
import seaborn as sns
import statsmodels.api as sm
import matplotlib.pyplot as plt
%matplotlib inline
import math
import statistics
import scipy.stats
import random

# Memanggil data

Sekarang mari kita panggil datanya menggunakan fungsi pada pandas dengan read_csv

In [48]:
df = pd.read_csv("nyc-rolling-sales.csv")
df

Unnamed: 0  BOROUGH   NEIGHBORHOOD  \
0               4        1  ALPHABET CITY   
1               5        1  ALPHABET CITY   
2               6        1  ALPHABET CITY   
3               7        1  ALPHABET CITY   
4               8        1  ALPHABET CITY   
...           ...      ...            ...   
84543        8409        5        WOODROW   
84544        8410        5        WOODROW   
84545        8411        5        WOODROW   
84546        8412        5        WOODROW   
84547        8413        5        WOODROW   

                           BUILDING CLASS CATEGORY TAX CLASS AT PRESENT  \
0      07 RENTALS - WALKUP APARTMENTS                                2A   
1      07 RENTALS - WALKUP APARTMENTS                                 2   
2      07 RENTALS - WALKUP APARTMENTS                                 2   
3      07 RENTALS - WALKUP APARTMENTS                                2B   
4      07 RENTALS - WALKUP APARTMENTS                                2A   
...                                            ...                  ...   
84543  02 TWO FAMILY DWELLINGS                                        1   
84544  02 TWO FAMILY DWELLINGS                                        1   
84545  02 TWO FAMILY DWELLINGS                                        1   
84546  22 STORE BUILDINGS                                             4   
84547  35 INDOOR PUBLIC AND CULTURAL FACILITIES                       4   

       BLOCK  LOT EASE-MENT BUILDING CLASS AT PRESENT                 ADDRESS  \
0        392    6                                  C2            153 AVENUE B   
1        399   26                                  C7   234 EAST 4TH   STREET   
2        399   39                                  C7   197 EAST 3RD   STREET   
3        402   21                                  C4     154 EAST 7TH STREET   
4        404   55                                  C2  301 EAST 10TH   STREET   
...      ...  ...       ...                       ...                     ...   
84543   7349   34                                  B9           37 QUAIL LANE   
84544   7349   78                                  B9        32 PHEASANT LANE   
84545   7351   60                                  B2        49 PITNEY AVENUE   
84546   7100   28                                  K6   2730 ARTHUR KILL ROAD   
84547   7105  679                                  P9       155 CLAY PIT ROAD   

       ... RESIDENTIAL UNITS  COMMERCIAL UNITS  TOTAL UNITS  LAND SQUARE FEET  \
0      ...                 5                 0            5              1633   
1      ...                28                 3           31              4616   
2      ...                16                 1           17              2212   
3      ...                10                 0           10              2272   
4      ...                 6                 0            6              2369   
...    ...               ...               ...          ...               ...   
84543  ...                 2                 0            2              2400   
84544  ...                 2                 0            2              2498   
84545  ...                 2                 0            2              4000   
84546  ...                 0                 7            7            208033   
84547  ...                 0                 1            1             10796   

       GROSS SQUARE FEET YEAR BUILT TAX CLASS AT TIME OF SALE  \
0                   6440       1900                         2   
1                  18690       1900                         2   
2                   7803       1900                         2   
3                   6794       1913                         2   
4                   4615       1900                         2   
...                  ...        ...                       ...   
84543               2575       1998                         1   
84544               2377       1998                         1   
84545               1496       192

# Pengenalan Data

Setelah kita panggil datanya, mari kita kenali datanya, mulai dari tipe datanya,, jumlah datanya dan lainnya

In [49]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 84548 entries, 0 to 84547
Data columns (total 22 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Unnamed: 0                      84548 non-null  int64 
 1   BOROUGH                         84548 non-null  int64 
 2   NEIGHBORHOOD                    84548 non-null  object
 3   BUILDING CLASS CATEGORY         84548 non-null  object
 4   TAX CLASS AT PRESENT            84548 non-null  object
 5   BLOCK                           84548 non-null  int64 
 6   LOT                             84548 non-null  int64 
 7   EASE-MENT                       84548 non-null  object
 8   BUILDING CLASS AT PRESENT       84548 non-null  object
 9   ADDRESS                         84548 non-null  object
 10  APARTMENT NUMBER                84548 non-null  object
 11  ZIP CODE                        84548 non-null  int64 
 12  RESIDENTIAL UNITS               84548 non-null

In [50]:
df.shape

(84548, 22)

In [51]:
df.isna().sum()

Unnamed: 0                        0
BOROUGH                           0
NEIGHBORHOOD                      0
BUILDING CLASS CATEGORY           0
TAX CLASS AT PRESENT              0
BLOCK                             0
LOT                               0
EASE-MENT                         0
BUILDING CLASS AT PRESENT         0
ADDRESS                           0
APARTMENT NUMBER                  0
ZIP CODE                          0
RESIDENTIAL UNITS                 0
COMMERCIAL UNITS                  0
TOTAL UNITS                       0
LAND SQUARE FEET                  0
GROSS SQUARE FEET                 0
YEAR BUILT                        0
TAX CLASS AT TIME OF SALE         0
BUILDING CLASS AT TIME OF SALE    0
SALE PRICE                        0
SALE DATE                         0
dtype: int64

In [52]:
df.duplicated().sum()

0

In [53]:
df.describe()

Unnamed: 0       BOROUGH         BLOCK           LOT      ZIP CODE  \
count  84548.000000  84548.000000  84548.000000  84548.000000  84548.000000   
mean   10344.359878      2.998758   4237.218976    376.224015  10731.991614   
std     7151.779436      1.289790   3568.263407    658.136814   1290.879147   
min        4.000000      1.000000      1.000000      1.000000      0.000000   
25%     4231.000000      2.000000   1322.750000     22.000000  10305.000000   
50%     8942.000000      3.000000   3311.000000     50.000000  11209.000000   
75%    15987.250000      4.000000   6281.000000   1001.000000  11357.000000   
max    26739.000000      5.000000  16322.000000   9106.000000  11694.000000   

       RESIDENTIAL UNITS  COMMERCIAL UNITS   TOTAL UNITS    YEAR BUILT  \
count       84548.000000      84548.000000  84548.000000  84548.000000   
mean            2.025264          0.193559      2.249184   1789.322976   
std            16.721037          8.713183     18.972584    537.344993   
min             0.000000          0.000000      0.000000      0.000000   
25%             0.000000          0.000000      1.000000   1920.000000   
50%             1.000000          0.000000      1.000000   1940.000000   
75%             2.000000          0.000000      2.000000   1965.000000   
max          1844.000000       2261.000000   2261.000000   2017.000000   

       TAX CLASS AT TIME OF SALE  
count               84548.000000  
mean                    1.657485  
std                     0.819341  
min                     1.000000  
25%                     1.000000  
50%                     2.000000  
75%                     2.000000  
max                     4.000000

_Keterangan_

Dari data diatas kita bisa melihat tipe datanya. Akan tetapi ada beberapa tipe data yang tidak sesuai dengan varieblnya. Lalu jumlah datanya sebesar 84548 rows × 22 columns. Tidak ditemukan data yang missing value dan duplicated

# Data Cleansing dan Preparation Data

In [54]:
df

Unnamed: 0  BOROUGH   NEIGHBORHOOD  \
0               4        1  ALPHABET CITY   
1               5        1  ALPHABET CITY   
2               6        1  ALPHABET CITY   
3               7        1  ALPHABET CITY   
4               8        1  ALPHABET CITY   
...           ...      ...            ...   
84543        8409        5        WOODROW   
84544        8410        5        WOODROW   
84545        8411        5        WOODROW   
84546        8412        5        WOODROW   
84547        8413        5        WOODROW   

                           BUILDING CLASS CATEGORY TAX CLASS AT PRESENT  \
0      07 RENTALS - WALKUP APARTMENTS                                2A   
1      07 RENTALS - WALKUP APARTMENTS                                 2   
2      07 RENTALS - WALKUP APARTMENTS                                 2   
3      07 RENTALS - WALKUP APARTMENTS                                2B   
4      07 RENTALS - WALKUP APARTMENTS                                2A   
...                                            ...                  ...   
84543  02 TWO FAMILY DWELLINGS                                        1   
84544  02 TWO FAMILY DWELLINGS                                        1   
84545  02 TWO FAMILY DWELLINGS                                        1   
84546  22 STORE BUILDINGS                                             4   
84547  35 INDOOR PUBLIC AND CULTURAL FACILITIES                       4   

       BLOCK  LOT EASE-MENT BUILDING CLASS AT PRESENT                 ADDRESS  \
0        392    6                                  C2            153 AVENUE B   
1        399   26                                  C7   234 EAST 4TH   STREET   
2        399   39                                  C7   197 EAST 3RD   STREET   
3        402   21                                  C4     154 EAST 7TH STREET   
4        404   55                                  C2  301 EAST 10TH   STREET   
...      ...  ...       ...                       ...                     ...   
84543   7349   34                                  B9           37 QUAIL LANE   
84544   7349   78                                  B9        32 PHEASANT LANE   
84545   7351   60                                  B2        49 PITNEY AVENUE   
84546   7100   28                                  K6   2730 ARTHUR KILL ROAD   
84547   7105  679                                  P9       155 CLAY PIT ROAD   

       ... RESIDENTIAL UNITS  COMMERCIAL UNITS  TOTAL UNITS  LAND SQUARE FEET  \
0      ...                 5                 0            5              1633   
1      ...                28                 3           31              4616   
2      ...                16                 1           17              2212   
3      ...                10                 0           10              2272   
4      ...                 6                 0            6              2369   
...    ...               ...               ...          ...               ...   
84543  ...                 2                 0            2              2400   
84544  ...                 2                 0            2              2498   
84545  ...                 2                 0            2              4000   
84546  ...                 0                 7            7            208033   
84547  ...                 0                 1            1             10796   

       GROSS SQUARE FEET YEAR BUILT TAX CLASS AT TIME OF SALE  \
0                   6440       1900                         2   
1                  18690       1900                         2   
2                   7803       1900                         2   
3                   6794       1913                         2   
4                   4615       1900                         2   
...                  ...        ...                       ...   
84543               2575       1998                         1   
84544               2377       1998                         1   
84545               1496       192

Pertama kita akan mengubah nama borough menjadi nama area

In [55]:
df.rename(columns={"BOROUGH":"AREA"},inplace=True)

Setelah mengubah nama kolom borough menjadi area, sekarang kita mengubah isi data pada kolom borough dengan beberapa kategori

AREA berisi kode properti
1. Manhattan
2. Bronx
3. Brooklyn 
4. Queens
5. Staten Island

In [56]:
df['AREA'] = df.AREA.replace({1: "Manhattan", 2: "Bronx", 3: "Brooklyn", 4: "Queens", 5: "Staten Islands"})

In [57]:
df

Unnamed: 0            AREA   NEIGHBORHOOD  \
0               4       Manhattan  ALPHABET CITY   
1               5       Manhattan  ALPHABET CITY   
2               6       Manhattan  ALPHABET CITY   
3               7       Manhattan  ALPHABET CITY   
4               8       Manhattan  ALPHABET CITY   
...           ...             ...            ...   
84543        8409  Staten Islands        WOODROW   
84544        8410  Staten Islands        WOODROW   
84545        8411  Staten Islands        WOODROW   
84546        8412  Staten Islands        WOODROW   
84547        8413  Staten Islands        WOODROW   

                           BUILDING CLASS CATEGORY TAX CLASS AT PRESENT  \
0      07 RENTALS - WALKUP APARTMENTS                                2A   
1      07 RENTALS - WALKUP APARTMENTS                                 2   
2      07 RENTALS - WALKUP APARTMENTS                                 2   
3      07 RENTALS - WALKUP APARTMENTS                                2B   
4      07 RENTALS - WALKUP APARTMENTS                                2A   
...                                            ...                  ...   
84543  02 TWO FAMILY DWELLINGS                                        1   
84544  02 TWO FAMILY DWELLINGS                                        1   
84545  02 TWO FAMILY DWELLINGS                                        1   
84546  22 STORE BUILDINGS                                             4   
84547  35 INDOOR PUBLIC AND CULTURAL FACILITIES                       4   

       BLOCK  LOT EASE-MENT BUILDING CLASS AT PRESENT                 ADDRESS  \
0        392    6                                  C2            153 AVENUE B   
1        399   26                                  C7   234 EAST 4TH   STREET   
2        399   39                                  C7   197 EAST 3RD   STREET   
3        402   21                                  C4     154 EAST 7TH STREET   
4        404   55                                  C2  301 EAST 10TH   STREET   
...      ...  ...       ...                       ...                     ...   
84543   7349   34                                  B9           37 QUAIL LANE   
84544   7349   78                                  B9        32 PHEASANT LANE   
84545   7351   60                                  B2        49 PITNEY AVENUE   
84546   7100   28                                  K6   2730 ARTHUR KILL ROAD   
84547   7105  679                                  P9       155 CLAY PIT ROAD   

       ... RESIDENTIAL UNITS  COMMERCIAL UNITS  TOTAL UNITS  LAND SQUARE FEET  \
0      ...                 5                 0            5              1633   
1      ...                28                 3           31              4616   
2      ...                16                 1           17              2212   
3      ...                10                 0           10              2272   
4      ...                 6                 0            6              2369   
...    ...               ...               ...          ...               ...   
84543  ...                 2                 0            2              2400   
84544  ...                 2                 0            2              2498   
84545  ...                 2                 0            2              4000   
84546  ...                 0                 7            7            208033   
84547  ...                 0                 1            1             10796   

       GROSS SQUARE FEET YEAR BUILT TAX CLASS AT TIME OF SALE  \
0                   6440       1900                         2   
1                  18690       1900                         2   
2                   7803       1900                         2   
3                   6794       1913                         2   
4                   4615       1900                         2   
...                  ...        ...                       ...   
84543               2575       1998                         1   
84544          

Setelah mengubah dengan sesuai nama areanya, sekarang kita mengahpus kolom yang tidak dibutuhkan menggunakan drop kolom. Kolom yang tidak dipakai dalah Unnamed: 0 dan EASE_MEANT

In [58]:
df.drop(["Unnamed: 0","EASE-MENT"],axis=1,inplace=True)
df

AREA   NEIGHBORHOOD  \
0           Manhattan  ALPHABET CITY   
1           Manhattan  ALPHABET CITY   
2           Manhattan  ALPHABET CITY   
3           Manhattan  ALPHABET CITY   
4           Manhattan  ALPHABET CITY   
...               ...            ...   
84543  Staten Islands        WOODROW   
84544  Staten Islands        WOODROW   
84545  Staten Islands        WOODROW   
84546  Staten Islands        WOODROW   
84547  Staten Islands        WOODROW   

                           BUILDING CLASS CATEGORY TAX CLASS AT PRESENT  \
0      07 RENTALS - WALKUP APARTMENTS                                2A   
1      07 RENTALS - WALKUP APARTMENTS                                 2   
2      07 RENTALS - WALKUP APARTMENTS                                 2   
3      07 RENTALS - WALKUP APARTMENTS                                2B   
4      07 RENTALS - WALKUP APARTMENTS                                2A   
...                                            ...                  ...   
84543  02 TWO FAMILY DWELLINGS                                        1   
84544  02 TWO FAMILY DWELLINGS                                        1   
84545  02 TWO FAMILY DWELLINGS                                        1   
84546  22 STORE BUILDINGS                                             4   
84547  35 INDOOR PUBLIC AND CULTURAL FACILITIES                       4   

       BLOCK  LOT BUILDING CLASS AT PRESENT                 ADDRESS  \
0        392    6                        C2            153 AVENUE B   
1        399   26                        C7   234 EAST 4TH   STREET   
2        399   39                        C7   197 EAST 3RD   STREET   
3        402   21                        C4     154 EAST 7TH STREET   
4        404   55                        C2  301 EAST 10TH   STREET   
...      ...  ...                       ...                     ...   
84543   7349   34                        B9           37 QUAIL LANE   
84544   7349   78                        B9        32 PHEASANT LANE   
84545   7351   60                        B2        49 PITNEY AVENUE   
84546   7100   28                        K6   2730 ARTHUR KILL ROAD   
84547   7105  679                        P9       155 CLAY PIT ROAD   

      APARTMENT NUMBER  ZIP CODE  RESIDENTIAL UNITS  COMMERCIAL UNITS  \
0                          10009                  5                 0   
1                          10009                 28                 3   
2                          10009                 16                 1   
3                          10009                 10                 0   
4                          10009                  6                 0   
...                ...       ...                ...               ...   
84543                      10309                  2                 0   
84544                      10309                  2                 0   
84545                      10309                  2                 0   
84546                      10309                  0                 7   
84547                      10309                  0                 1   

       TOTAL UNITS LAND SQUARE FEET GROSS SQUARE FEET  YEAR BUILT  \
0                5             1633              6440        1900   
1               31             4616             18690        1900   
2               17             2212              7803        1900   
3               10             2272              6794        1913   
4                6             2369              4615        1900   
...            ...              ...               ...         ...   
84543            2             2400              2575        1998   
84544            2             2498              2377        1998   
84545            2             4000              1496        1925   
84546            7           208033             64117        2001   
84547            1            10796              2400        2006   

       TAX CLASS AT TIME OF SALE BUILDING CLASS AT TIME OF SALE SALE PRICE  

Mari Kita cek data yang duplikat

In [59]:
df.duplicated().sum()

765

DAta yang duplicated ada 765. Data tersebut tidak lebih dari 10%. Mari kita delete saja data nya agar tidak terjadi kesalahan pada analisis

In [60]:
df.drop_duplicates(inplace=True)

In [61]:
df.duplicated().sum()

0

Sekarang mari kita ubah tipe data yang tidak sesuai dengan variabelnya

In [62]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 83783 entries, 0 to 84547
Data columns (total 20 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   AREA                            83783 non-null  object
 1   NEIGHBORHOOD                    83783 non-null  object
 2   BUILDING CLASS CATEGORY         83783 non-null  object
 3   TAX CLASS AT PRESENT            83783 non-null  object
 4   BLOCK                           83783 non-null  int64 
 5   LOT                             83783 non-null  int64 
 6   BUILDING CLASS AT PRESENT       83783 non-null  object
 7   ADDRESS                         83783 non-null  object
 8   APARTMENT NUMBER                83783 non-null  object
 9   ZIP CODE                        83783 non-null  int64 
 10  RESIDENTIAL UNITS               83783 non-null  int64 
 11  COMMERCIAL UNITS                83783 non-null  int64 
 12  TOTAL UNITS                     83783 non-null

Sekarang mari kita ubah untuk kolom 13,14,18 keadalam int

Kita gunakan perulangan untuk mengecek kolom untuk megubhanya menjadi dalam bentuk numrik, Nanti ada errors="coerce"  yang gunanya untuk membuat nilai yang tidak bisa dikonver jadi NAn

In [63]:
numerical_ = ["LAND SQUARE FEET","GROSS SQUARE FEET","SALE PRICE"]

for column in numerical_:
    df[column]=pd.to_numeric(df[column],errors="coerce")

In [64]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 83783 entries, 0 to 84547
Data columns (total 20 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   AREA                            83783 non-null  object 
 1   NEIGHBORHOOD                    83783 non-null  object 
 2   BUILDING CLASS CATEGORY         83783 non-null  object 
 3   TAX CLASS AT PRESENT            83783 non-null  object 
 4   BLOCK                           83783 non-null  int64  
 5   LOT                             83783 non-null  int64  
 6   BUILDING CLASS AT PRESENT       83783 non-null  object 
 7   ADDRESS                         83783 non-null  object 
 8   APARTMENT NUMBER                83783 non-null  object 
 9   ZIP CODE                        83783 non-null  int64  
 10  RESIDENTIAL UNITS               83783 non-null  int64  
 11  COMMERCIAL UNITS                83783 non-null  int64  
 12  TOTAL UNITS                     

Sekarang kita ubah 'NEIGHBORHOOD', 'BUILDING CLASS CATEGORY', 'TAX CLASS AT PRESENT', 'BUILDING CLASS AT TIME OF SALE', 'TAX CLASS AT TIME OF SALE', 'AREA' menjadi category

In [65]:
category_ = ['NEIGHBORHOOD', 'BUILDING CLASS CATEGORY', 'TAX CLASS AT PRESENT', 'BUILDING CLASS AT TIME OF SALE', 'TAX CLASS AT TIME OF SALE', 'AREA']
for column in category_:
    df[column]=df[column].astype("category")

In [66]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 83783 entries, 0 to 84547
Data columns (total 20 columns):
 #   Column                          Non-Null Count  Dtype   
---  ------                          --------------  -----   
 0   AREA                            83783 non-null  category
 1   NEIGHBORHOOD                    83783 non-null  category
 2   BUILDING CLASS CATEGORY         83783 non-null  category
 3   TAX CLASS AT PRESENT            83783 non-null  category
 4   BLOCK                           83783 non-null  int64   
 5   LOT                             83783 non-null  int64   
 6   BUILDING CLASS AT PRESENT       83783 non-null  object  
 7   ADDRESS                         83783 non-null  object  
 8   APARTMENT NUMBER                83783 non-null  object  
 9   ZIP CODE                        83783 non-null  int64   
 10  RESIDENTIAL UNITS               83783 non-null  int64   
 11  COMMERCIAL UNITS                83783 non-null  int64   
 12  TOTAL UNITS       

Guna dari category adalah digunakan untuk menyimpan data kategorikal atau data yang terdiri dari sejumlah kategori. Ini digunakan untuk mengurangi penggunaan memori, meningkatkan kecepatan pemrosesan data.

================================================

Mari kita ubah sale date menjadi timestamp

In [67]:
df["SALE DATE"] = pd.to_datetime(df["SALE DATE"], errors='coerce')

In [68]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 83783 entries, 0 to 84547
Data columns (total 20 columns):
 #   Column                          Non-Null Count  Dtype         
---  ------                          --------------  -----         
 0   AREA                            83783 non-null  category      
 1   NEIGHBORHOOD                    83783 non-null  category      
 2   BUILDING CLASS CATEGORY         83783 non-null  category      
 3   TAX CLASS AT PRESENT            83783 non-null  category      
 4   BLOCK                           83783 non-null  int64         
 5   LOT                             83783 non-null  int64         
 6   BUILDING CLASS AT PRESENT       83783 non-null  object        
 7   ADDRESS                         83783 non-null  object        
 8   APARTMENT NUMBER                83783 non-null  object        
 9   ZIP CODE                        83783 non-null  int64         
 10  RESIDENTIAL UNITS               83783 non-null  int64         
 11  CO

Stelah semua tipe data yang saya inginkan sudah sesuai, mari kita cek lagi apakah ada missing value dari konvert tipe data tsb

In [69]:
df.isna().sum()

AREA                                  0
NEIGHBORHOOD                          0
BUILDING CLASS CATEGORY               0
TAX CLASS AT PRESENT                  0
BLOCK                                 0
LOT                                   0
BUILDING CLASS AT PRESENT             0
ADDRESS                               0
APARTMENT NUMBER                      0
ZIP CODE                              0
RESIDENTIAL UNITS                     0
COMMERCIAL UNITS                      0
TOTAL UNITS                           0
LAND SQUARE FEET                  26054
GROSS SQUARE FEET                 27385
YEAR BUILT                            0
TAX CLASS AT TIME OF SALE             0
BUILDING CLASS AT TIME OF SALE        0
SALE PRICE                        14176
SALE DATE                             0
dtype: int64

Ternyata terdapat missing value pada Land Square, Gross Square, Sale Price

mari kita gantikan missing value tsbt dengan nilai mean

In [70]:
df.describe()

BLOCK           LOT      ZIP CODE  RESIDENTIAL UNITS  \
count  83783.000000  83783.000000  83783.000000       83783.000000   
mean    4242.903155    376.137415  10733.236575           1.999976   
std     3568.789482    658.021286   1289.764300          16.569118   
min        1.000000      1.000000      0.000000           0.000000   
25%     1326.000000     22.000000  10305.000000           0.000000   
50%     3319.000000     50.000000  11209.000000           1.000000   
75%     6295.000000   1001.000000  11357.000000           2.000000   
max    16322.000000   9106.000000  11694.000000        1844.000000   

       COMMERCIAL UNITS   TOTAL UNITS  LAND SQUARE FEET  GROSS SQUARE FEET  \
count      83783.000000  83783.000000      5.772900e+04       5.639800e+04   
mean           0.183677      2.213504      3.854481e+03       3.894416e+03   
std            8.576582     18.777727      4.154737e+04       3.483726e+04   
min            0.000000      0.000000      0.000000e+00       0.000000e+00   
25%            0.000000      1.000000      1.643000e+03       1.040000e+03   
50%            0.000000      1.000000      2.314000e+03       1.680000e+03   
75%            0.000000      2.000000      3.500000e+03       2.552000e+03   
max         2261.000000   2261.000000      4.252327e+06       3.750565e+06   

         YEAR BUILT    SALE PRICE  
count  83783.000000  6.960700e+04  
mean    1789.814318  1.280703e+06  
std      536.565051  1.143036e+07  
min        0.000000  0.000000e+00  
25%     1920.000000  2.300000e+05  
50%     1940.000000  5.330000e+05  
75%     1965.000000  9.500000e+05  
max     2017.000000  2.210000e+09

In [71]:
df['LAND SQUARE FEET'].fillna(df['LAND SQUARE FEET'].mean(), inplace=True)
df['GROSS SQUARE FEET'].fillna(df['GROSS SQUARE FEET'].mean(), inplace=True)
df['SALE PRICE'].fillna(df['SALE PRICE'].mean(), inplace=True)

In [72]:
df

AREA   NEIGHBORHOOD  \
0           Manhattan  ALPHABET CITY   
1           Manhattan  ALPHABET CITY   
2           Manhattan  ALPHABET CITY   
3           Manhattan  ALPHABET CITY   
4           Manhattan  ALPHABET CITY   
...               ...            ...   
84543  Staten Islands        WOODROW   
84544  Staten Islands        WOODROW   
84545  Staten Islands        WOODROW   
84546  Staten Islands        WOODROW   
84547  Staten Islands        WOODROW   

                           BUILDING CLASS CATEGORY TAX CLASS AT PRESENT  \
0      07 RENTALS - WALKUP APARTMENTS                                2A   
1      07 RENTALS - WALKUP APARTMENTS                                 2   
2      07 RENTALS - WALKUP APARTMENTS                                 2   
3      07 RENTALS - WALKUP APARTMENTS                                2B   
4      07 RENTALS - WALKUP APARTMENTS                                2A   
...                                            ...                  ...   
84543  02 TWO FAMILY DWELLINGS                                        1   
84544  02 TWO FAMILY DWELLINGS                                        1   
84545  02 TWO FAMILY DWELLINGS                                        1   
84546  22 STORE BUILDINGS                                             4   
84547  35 INDOOR PUBLIC AND CULTURAL FACILITIES                       4   

       BLOCK  LOT BUILDING CLASS AT PRESENT                 ADDRESS  \
0        392    6                        C2            153 AVENUE B   
1        399   26                        C7   234 EAST 4TH   STREET   
2        399   39                        C7   197 EAST 3RD   STREET   
3        402   21                        C4     154 EAST 7TH STREET   
4        404   55                        C2  301 EAST 10TH   STREET   
...      ...  ...                       ...                     ...   
84543   7349   34                        B9           37 QUAIL LANE   
84544   7349   78                        B9        32 PHEASANT LANE   
84545   7351   60                        B2        49 PITNEY AVENUE   
84546   7100   28                        K6   2730 ARTHUR KILL ROAD   
84547   7105  679                        P9       155 CLAY PIT ROAD   

      APARTMENT NUMBER  ZIP CODE  RESIDENTIAL UNITS  COMMERCIAL UNITS  \
0                          10009                  5                 0   
1                          10009                 28                 3   
2                          10009                 16                 1   
3                          10009                 10                 0   
4                          10009                  6                 0   
...                ...       ...                ...               ...   
84543                      10309                  2                 0   
84544                      10309                  2                 0   
84545                      10309                  2                 0   
84546                      10309                  0                 7   
84547                      10309                  0                 1   

       TOTAL UNITS  LAND SQUARE FEET  GROSS SQUARE FEET  YEAR BUILT  \
0                5            1633.0             6440.0        1900   
1               31            4616.0            18690.0        1900   
2               17            2212.0             7803.0        1900   
3               10            2272.0             6794.0        1913   
4                6            2369.0             4615.0        1900   
...            ...               ...                ...         ...   
84543            2            2400.0             2575.0        1998   
84544            2            2498.0             2377.0        1998   
84545            2            4000.0             1496.0        1925   
84546            7          208033.0            64117.0        2001   
84547            1           10796.0             2400.0        2006   

      TAX CLASS AT TIME OF SALE BUILDING CLASS AT TI

In [73]:
df.isna().sum()

AREA                              0
NEIGHBORHOOD                      0
BUILDING CLASS CATEGORY           0
TAX CLASS AT PRESENT              0
BLOCK                             0
LOT                               0
BUILDING CLASS AT PRESENT         0
ADDRESS                           0
APARTMENT NUMBER                  0
ZIP CODE                          0
RESIDENTIAL UNITS                 0
COMMERCIAL UNITS                  0
TOTAL UNITS                       0
LAND SQUARE FEET                  0
GROSS SQUARE FEET                 0
YEAR BUILT                        0
TAX CLASS AT TIME OF SALE         0
BUILDING CLASS AT TIME OF SALE    0
SALE PRICE                        0
SALE DATE                         0
dtype: int64

# Measure of Conten

Disini kita akan mencari Nilai Mean ( rata rata ), Nilai Median ( nilai tengah), dan Nilai Modus ( nilai sering muncul) pada 5 tahun terahkir

## Rumusan Masalah
> Berapa nilai rata rata, nilai median pada penjualan unit, luas , dan harga di 5 tahun terahkir? (mean)

> Berapa nilai nilai median pada penjualan unit, luas , dan harga di 5 tahun terahkir? (median)

> Nilai Mayoritas manakan yang sering muncul terpacu pada daerah? (Mode)

### Mean

Kita akan mencari Mean dari setiap tahunnya, mulai dari unit, luas, dan harga dari 5 tahun terahkir

In [74]:
#mean Pandas
a = df.groupby(["YEAR BUILT"])[['RESIDENTIAL UNITS','COMMERCIAL UNITS','TOTAL UNITS','LAND SQUARE FEET','GROSS SQUARE FEET','SALE PRICE']].mean()
a.tail(5)

RESIDENTIAL UNITS  COMMERCIAL UNITS  TOTAL UNITS  \
YEAR BUILT                                                     
2013                 2.160377          0.071429     2.346361   
2014                 2.226016          0.519512     2.886179   
2015                 1.965870          0.561092     2.622526   
2016                 4.092172          0.051768     4.208333   
2017                 2.000000        141.166667   143.166667   

            LAND SQUARE FEET  GROSS SQUARE FEET    SALE PRICE  
YEAR BUILT                                                     
2013             3585.912404        4481.947674  2.882327e+06  
2014             2533.388581        4441.436948  2.204815e+06  
2015             2256.184251        2987.842885  1.453862e+06  
2016             3310.539724        2495.027867  8.763765e+05  
2017             5237.166667        1529.305271  5.719010e+05

#### _KETERANGAN nilai Mean_
1. Pada tahun 2013, 
   1. Residental Units memiliki nilai rata rata 2.160377 
   2. Commercial Units memiliki rata rata 0.071429	 
   3. Total Units memiliki rata rata 2.346361 
   4. Land Square Feet  memiliki rata rata 3585.912404 
   5. Gross Square Feet memiliki rata rata 4481.947674 
   6. Sale Price dengan rata rata 2.882327e+06


2. Pada tahun 2014, 
   1. Residental Units memiliki nilai rata rata 2.226016 
   2. Commercial Units memiliki rata rata 0.519512 
   3. Total Units memiliki rata rata 2.886179 
   4. Land Square Feet  memiliki rata rata 2533.388581 
   5. Gross Square Feet memiliki rata rata 4441.436948 
   6. Sale Price dengan rata rata 2.204815e+06


3. Pada tahun 2015, 
   1. Residental Units memiliki nilai rata rata 1.965870 
   2. Commercial Units memiliki rata rata 0.561092 
   3. Total Units memiliki rata rata 2.622526 
   4. Land Square Feet  memiliki rata rata 2256.184251 
   5. Gross Square Feet memiliki rata rata 2987.842885 
   6. Sale Price dengan rata rata 1.453862e+06


4. Pada tahun 2016, 
   1. Residental Units memiliki nilai rata rata 4.092172 
   2. Commercial Units memiliki rata rata 0.051768 
   3. Total Units memiliki rata rata 4.208333 
   4. Land Square Feet  memiliki rata rata 3310.539724 
   5. Gross Square Feet memiliki rata rata 2495.027867 
   6. Sale Price dengan rata rata 8.763765e+05


5. Pada tahun 2017, 
   1. Residental Units memiliki nilai rata rata 2.000000 
   2. Commercial Units memiliki rata rata 141.166667 
   3. Total Units memiliki rata rata 143.166667 
   4. Land Square Feet  memiliki rata rata 5237.166667 
   5. Gross Square Feet memiliki rata rata 1529.305271 
   6. Sale Price dengan rata rata 5.719010e+05

### Median

In [75]:
#median dengan pandas
b = df.groupby(["YEAR BUILT"])[['RESIDENTIAL UNITS','COMMERCIAL UNITS','TOTAL UNITS','LAND SQUARE FEET','GROSS SQUARE FEET','SALE PRICE']].median()
b.tail(5)

RESIDENTIAL UNITS  COMMERCIAL UNITS  TOTAL UNITS  \
YEAR BUILT                                                     
2013                      1.0               0.0          1.0   
2014                      1.0               0.0          1.0   
2015                      1.0               0.0          1.0   
2016                      1.0               0.0          1.0   
2017                      0.0               1.0          6.5   

            LAND SQUARE FEET  GROSS SQUARE FEET    SALE PRICE  
YEAR BUILT                                                     
2013             3854.481179        3894.415813  9.941290e+05  
2014             3854.481179        3894.415813  1.280703e+06  
2015             2300.000000        2059.000000  8.858770e+05  
2016             3000.000000        2244.000000  7.203995e+05  
2017             3367.500000         693.500000  4.350000e+05

#### _KETERANGAN Nilai Median Pada tabel_
1. Pada tahun 2013, 
   1. Residental Units memiliki nilai median 1.0 
   2. Commercial Units memiliki median 0.0		 
   3. Total Units memiliki median 1.0	 
   4. Land Square Feet  memiliki median 3854.481179	
   5. Gross Square Feet memiliki median 3894.415813 
   6. Sale Price dengan median 9.941290e+05


2. Pada tahun 2014, 
   1. Residental Units memiliki nilai median 1.0
   2. Commercial Units memiliki median 0.0		 
   3. Total Units memiliki median 1.0	
   4. Land Square Feet  memiliki median 3854.481179	 
   5. Gross Square Feet memiliki median 3894.415813 
   6. Sale Price dengan median 1.280703e+06


3. Pada tahun 2015, 
   1. Residental Units memiliki nilai median 1.0
   2. Commercial Units memiliki median 0.0		 
   3. Total Units memiliki median 1.0	 
   4. Land Square Feet  memiliki median 2300.000000 
   5. Gross Square Feet memiliki median 2059.000000 
   6. Sale Price dengan median 8.858770e+05


4. Pada tahun 2016, 
   1. Residental Units memiliki nilai median 1.0
   2. Commercial Units memiliki median 0.0		 
   3. Total Units memiliki median 1.0	 
   4. Land Square Feet  memiliki median 3000.000000	 
   5. Gross Square Feet memiliki median 2244.000000 
   6. Sale Price dengan median 7.203995e+05


5. Pada tahun 2017, 
   1. Residental Units memiliki nilai median 0.0 
   2. Commercial Units memiliki median 1.0	 
   3. Total Units memiliki median 6.5 
   4. Land Square Feet  memiliki median 3367.500000	 
   5. Gross Square Feet memiliki median 693.500000 
   6. Sale Price dengan median 4.350000e+05

### Mode

In [76]:
#mencari mode dengan pandas
df.groupby('AREA').apply(pd.DataFrame.mode).reset_index(drop=True)

AREA             NEIGHBORHOOD  \
0           Bronx                RIVERDALE   
1        Brooklyn       BEDFORD STUYVESANT   
2       Manhattan  UPPER EAST SIDE (59-79)   
3          Queens           FLUSHING-NORTH   
4  Staten Islands              GREAT KILLS   

                       BUILDING CLASS CATEGORY TAX CLASS AT PRESENT  BLOCK  \
0  02 TWO FAMILY DWELLINGS                                        1   3944   
1  02 TWO FAMILY DWELLINGS                                        1   2135   
2  13 CONDOS - ELEVATOR APARTMENTS                                2     16   
3  01 ONE FAMILY DWELLINGS                                        1   5066   
4  01 ONE FAMILY DWELLINGS                                        1   2450   

   LOT BUILDING CLASS AT PRESENT                 ADDRESS APARTMENT NUMBER  \
0    1                        D4     2287 JOHNSON AVENUE                    
1    1                        R4         429 KENT AVENUE                    
2    1                        R4  169 WEST 95TH   STREET                    
3    1                        A1      131-05 40TH   ROAD                    
4    1                        A5   1235 FOREST HILL ROAD                    

   ZIP CODE  RESIDENTIAL UNITS  COMMERCIAL UNITS  TOTAL UNITS  \
0     10462                  1                 0            1   
1     11201                  1                 0            1   
2     10011                  0                 0            1   
3     11354                  1                 0            1   
4     10314                  1                 0            1   

   LAND SQUARE FEET  GROSS SQUARE FEET  YEAR BUILT TAX CLASS AT TIME OF SALE  \
0          0.000000           0.000000           0                         1   
1          0.000000           0.000000           0                         1   
2       3854.481179        3894.415813           0                         2   
3       3854.481179        3894.415813        1920                         1   
4       3854.481179        3894.415813        1970                         1   

  BUILDING CLASS AT TIME OF SALE    SALE PRICE  SALE DATE  
0                             D4  0.000000e+00 2017-06-29  
1                             R4  0.000000e+00 2016-10-06  
2                             R4  1.280703e+06 2017-08-07  
3                             A1  1.280703e+06 2016-11-10  
4                             A5  1.280703e+06 2017-03-01

#### _Keterangan Mode_

Dari data yang ditampilkan, dijelaskan bahwa 

1. Data yang sering muncul pada daerah **BRONX**
    1. **NEIGHBORHOOD** adalah _**RIVERDALE**_
    2. **BUILIDING CLASS CATEGORY** adalah _**02 TWO FAMILY DWELLINGS**_
    3. **TAX CALSS AT PRESENT** adalah _**1**_
    4. **BLOCK** adalah _**3944**_
    5. **LOT** adalah _**1**_
    6. **BUILIDING CLASS AT PRESENT** adalah _**D4**_
    7. **ADDRES** adalah _**2287 JOHNSON AVENUE**_
    8. **ZIP CODE** adalah _**10462**_
    9. **RESIDENTAL UNITS** adalah _**1**_
    11. **COMMERCIAL UNITS** adalah _**0**_
    12. **TOTAL UNITS** adalah _**1**_
    13. **LAND SQUARE FEET** adalah _**0.000000**_
    14. **GROSS SQUARE FEET** adalah _**0.000000**_
    15. **YEAR BUILT** adalah _**0**_
    16. **TAX CLASS AT TIME OF SALE** adalah _**1**_
    17. **BUILDING CLASS AT TIME OF SALE** adalah _**D4**_
    18. **SALE PRICE** adalah _**0.000000e+00**_
    19. **SALKE DATE** adalah _**2017-06-29**_
    
    
2. Data yang sering muncul pada daerah **BROKLYN**
    1. **NEIGHBORHOOD** adalah _**BEDFORD STUYVESANT**_
    2. **BUILIDING CLASS CATEGORY** adalah _**02 TWO FAMILY DWELLINGS**_
    3. **TAX CALSS AT PRESENT** adalah _**1**_
    4. **BLOCK** adalah _**2135**_
    5. **LOT** adalah _**1**_
    6. **BUILIDING CLASS AT PRESENT** adalah _**R4**_
    7. **ADDRES** adalah _**429 KENT AVENUE**_
    8. **ZIP CODE** adalah _**11201**_
    9. **RESIDENTAL UNITS** adalah _**1**_
    11. **COMMERCIAL UNITS** adalah _**0**_
    12. **TOTAL UNITS** adalah _**1**_
    13. **LAND SQUARE FEET** adalah _**0.000000**_
    14. **GROSS SQUARE FEET** adalah _**0.000000**_
    15. **YEAR BUILT** adalah _**0**_
    16. **TAX CLASS AT TIME OF SALE** adalah _**1**_
    17. **BUILDING CLASS AT TIME OF SALE** adalah _**R4**_
    18. **SALE PRICE** adalah _**0.000000e+00**_
    19. **SALKE DATE** adalah _**2016-10-06**_
    
    
3. Data yang sering muncul pada daerah **MANHATTAN**
    1. **NEIGHBORHOOD** adalah _**UPPER EAST SIDE (59-79)**_
    2. **BUILIDING CLASS CATEGORY** adalah _**13 CONDOS - ELEVATOR APARTMENTS**_
    3. **TAX CALSS AT PRESENT** adalah _**2**_
    4. **BLOCK** adalah _**16**_
    5. **LOT** adalah _**1**_
    6. **BUILIDING CLASS AT PRESENT** adalah _**R4**_
    7. **ADDRES** adalah _**169 WEST 95TH STREET**_
    8. **ZIP CODE** adalah _**10011**_
    9. **RESIDENTAL UNITS** adalah _**0**_
    11. **COMMERCIAL UNITS** adalah _**0**_
    12. **TOTAL UNITS** adalah _**1**_
    13. **LAND SQUARE FEET** adalah _**3854.481179**_
    14. **GROSS SQUARE FEET** adalah _**3894.415813**_
    15. **YEAR BUILT** adalah _**0**_
    16. **TAX CLASS AT TIME OF SALE** adalah _**2**_
    17. **BUILDING CLASS AT TIME OF SALE** adalah _**R4**_
    18. **SALE PRICE** adalah _**1.280703e+06**_
    19. **SALKE DATE** adalah _**2017-08-07**_
    
    
4. Data yang sering muncul pada daerah **QUEENS**
    1. **NEIGHBORHOOD** adalah _**FLUSHING-NORTH**_
    2. **BUILIDING CLASS CATEGORY** adalah _**01 ONE FAMILY DWELLINGS**_
    3. **TAX CALSS AT PRESENT** adalah _**1**_
    4. **BLOCK** adalah _**5066**_
    5. **LOT** adalah _**1**_
    6. **BUILIDING CLASS AT PRESENT** adalah _**A1**_
    7. **ADDRES** adalah _**131-05 40TH ROAD**_
    8. **ZIP CODE** adalah _**11354**_
    9. **RESIDENTAL UNITS** adalah _**1**_
    11. **COMMERCIAL UNITS** adalah _**0**_
    12. **TOTAL UNITS** adalah _**1**_
    13. **LAND SQUARE FEET** adalah _**3854.481179**_
    14. **GROSS SQUARE FEET** adalah _**3894.415813**_
    15. **YEAR BUILT** adalah _**1920**_
    16. **TAX CLASS AT TIME OF SALE** adalah _**1**_
    17. **BUILDING CLASS AT TIME OF SALE** adalah _**A1**_
    18. **SALE PRICE** adalah _**1.280703e+06**_
    19. **SALKE DATE** adalah _**2016-11-10**_
    
    
5. Data yang sering muncul pada daerah **STATEND ISLIND**
    1. **NEIGHBORHOOD** adalah _**GREAT KILLS**_
    2. **BUILIDING CLASS CATEGORY** adalah _**01 ONE FAMILY DWELLINGS**_
    3. **TAX CALSS AT PRESENT** adalah _**1**_
    4. **BLOCK** adalah _**2450**_
    5. **LOT** adalah _**1**_
    6. **BUILIDING CLASS AT PRESENT** adalah _**A5**_
    7. **ADDRES** adalah _**1235 FOREST HILL ROAD**_
    8. **ZIP CODE** adalah _**10314**_
    9. **RESIDENTAL UNITS** adalah _**1**_
    11. **COMMERCIAL UNITS** adalah _**0**_
    12. **TOTAL UNITS** adalah _**1**_
    13. **LAND SQUARE FEET** adalah _**3854.481179**_
    14. **GROSS SQUARE FEET** adalah _**3894.415813**_
    15. **YEAR BUILT** adalah _**1970**_
    16. **TAX CLASS AT TIME OF SALE** adalah _**1**_
    17. **BUILDING CLASS AT TIME OF SALE** adalah _**A5**_
    18. **SALE PRICE** adalah _**1.280703e+06**_
    19. **SALKE DATE** adalah _**2017-03-01**_

# Measure of Spread

1. Range : Selisih antara nilai terbesar dan terkecil dalam suatu data.
2. Variance : Ukuran penyebaran yang mengukur seberapa jauh data tersebar dari nilai rata-rata. Varians merupakan rata-rata dari kuadrat selisih antara setiap data dengan nilai rata-rata.
3. Standard Deviation : Akar kuadrat dari varians. 

## Rumusan Masalah

1. Berapakah range units, luas, dan harga pada 5 tahun terahkir?
2. Berapakah Variance dari units, luas, dan harga pada 5 tahun terahkir?
3. Berapakah Standart Deviasi dari  dari units, luas, dan harga pada 5 tahun terahkir?

### Range 

In [77]:
#mencari range dengan padnas
range_data = df.groupby(["YEAR BUILT"])[['RESIDENTIAL UNITS','COMMERCIAL UNITS','TOTAL UNITS','LAND SQUARE FEET','GROSS SQUARE FEET','SALE PRICE']].agg(np.ptp)
range_data.tail(5)

RESIDENTIAL UNITS  COMMERCIAL UNITS  TOTAL UNITS  \
YEAR BUILT                                                     
2013                      233                25          234   
2014                      233               318          317   
2015                      233               254          253   
2016                      550                 4          551   
2017                       10               422          421   

            LAND SQUARE FEET  GROSS SQUARE FEET    SALE PRICE  
YEAR BUILT                                                     
2013                499600.0      348263.000000  6.773410e+07  
2014                 50000.0      533600.000000  2.033450e+08  
2015                 57936.0      167943.000000  1.740000e+08  
2016                110303.0      143700.000000  2.545625e+07  
2017                 12344.0        3894.415813  1.280703e+06

#### _Keterangan NIlai Range_
Data diatas adalah menampilkan range untuk kolom Residental Units, Comercial Units, Total Units, Lands Feet, Gross Square Feet , dan Sale Price pada 5 tahun terahkir

### Variance

In [78]:
#mencari variance dengan pandas
variance_data = df.groupby(["YEAR BUILT"])[['RESIDENTIAL UNITS','COMMERCIAL UNITS','TOTAL UNITS','LAND SQUARE FEET','GROSS SQUARE FEET','SALE PRICE']].var()
variance_data.tail(5)

RESIDENTIAL UNITS  COMMERCIAL UNITS   TOTAL UNITS  \
YEAR BUILT                                                      
2013               225.014730          0.895026    227.840736   
2014               204.445214        116.380010    320.302738   
2015               126.595829        131.940425    256.241979   
2016               888.771519          0.089605    891.540613   
2017                16.000000      47320.566667  46662.966667   

            LAND SQUARE FEET  GROSS SQUARE FEET    SALE PRICE  
YEAR BUILT                                                     
2013            3.407325e+08       3.508320e+08  3.254246e+13  
2014            6.425591e+06       5.207500e+08  5.147128e+13  
2015            8.997277e+06       1.026626e+08  2.339504e+13  
2016            4.465299e+07       3.368749e+07  1.782244e+12  
2017            3.287328e+07       3.644814e+06  4.149752e+11

#### _Keterangan NIlai Variance_
Data diatas adalah menampilkan Variance untuk kolom Residental Units, Comercial Units, Total Units, Lands Feet, Gross Square Feet , dan Sale Price pada 5 tahun terahkir

### Standart Deviasi

In [79]:
#mencari STD dengan pandas
std_data = df.groupby(["YEAR BUILT"])[['RESIDENTIAL UNITS','COMMERCIAL UNITS','TOTAL UNITS','LAND SQUARE FEET','GROSS SQUARE FEET','SALE PRICE']].std()
std_data.tail(5)

RESIDENTIAL UNITS  COMMERCIAL UNITS  TOTAL UNITS  \
YEAR BUILT                                                     
2013                15.000491          0.946058    15.094394   
2014                14.298434         10.787957    17.897004   
2015                11.251481         11.486532    16.007560   
2016                29.812271          0.299341    29.858677   
2017                 4.000000        217.532909   216.016126   

            LAND SQUARE FEET  GROSS SQUARE FEET    SALE PRICE  
YEAR BUILT                                                     
2013            18458.941103       18730.510997  5.704600e+06  
2014             2534.874886       22819.947917  7.174349e+06  
2015             2999.546162       10132.254153  4.836842e+06  
2016             6682.289595        5804.092786  1.335007e+06  
2017             5733.522632        1909.139616  6.441857e+05

#### _Keterangan NIlai Standart Deviasi_
Data diatas adalah menampilkan Variance untuk kolom Residental Units, Comercial Units, Total Units, Lands Feet, Gross Square Feet , dan Sale Price pada 5 tahun terahkir

# Probability Distribution

# Confident Intervlas

Rumusan Masalah

Berapa interval dari rata rata jual pada tahun 2014?

In [80]:
c = df.groupby('YEAR BUILT').agg({"SALE PRICE": [np.mean, np.std, np.size]})
c.tail(5)

SALE PRICE                    
                    mean           std  size
YEAR BUILT                                  
2013        2.882327e+06  5.704600e+06   742
2014        2.204815e+06  7.174349e+06  1230
2015        1.453862e+06  4.836842e+06  1465
2016        8.763765e+05  1.335007e+06   792
2017        5.719010e+05  6.441857e+05     6

Sekarang mari kita hitung rata rata 

In [81]:
mean_2014 = 2.204815e+06  # rata-rata harga properti dpada tahun 2014
std_2014 = 7.174349e+06 # standar deviasi harga properti pada tahun 2014
n = 1230 # jumlah properti di 2014
z = 1.96 # z-score berdasarkan tabel z
se = std_2014 / np.sqrt(n) # menghitung standart eror

# menghitung CI
lower = mean_2014 - z*se
upper = mean_2014 + z*se
print("The lower class boundary is", lower)
print("The upper class boundary is", upper)

The lower class boundary is 1803868.880130629
The upper class boundary is 2605761.119869371


berdasar hasil diatas, rata rata harga penjualan (sale price) pada tahun 2014 berada di antara 1803868.880130629 sampai 2605761.119869371 dengan peluang atau selang kepercayaan 30,77%

# Hipotesis Testing

Rumusan Masalah :

Dengan mempertimbangkan data NYC Property, apakah tahun 2017  memiliki nilai rata rata yang jual lebih tinggi dari tahun 2002?

H0: Rata-rata harga jual properti di tahun 2017 sama dengan rata-rata harga jual properti di tahun 2002.

H1: Rata-rata harga jual properti di tahun 2017 tidak sama dengan rata-rata harga jual properti di tahun 2002.

In [100]:
yb_17 = df[df["YEAR BUILT"] == 2017]

yb_02 = df[df["YEAR BUILT"] == 2002]

n_17 = len(yb_17)

mean_17 = yb_17["SALE PRICE"].mean()

std_17 = yb_17["SALE PRICE"].std()

(n_17, mean_17, std_17)

(6, 571901.0001005646, 644185.660577347)

In [101]:
n_02 = len(yb_02)

mean_02 = yb_02["SALE PRICE"].mean()

std_02 = yb_02["SALE PRICE"].std()

(n_02, mean_02, std_02)

(511, 1093619.6673980996, 1250592.0481180593)

In [102]:
sm.stats.ztest(yb_17["SALE PRICE"].dropna(),

           yb_02["SALE PRICE"].dropna(),

          alternative='two-sided')

(-1.0195673315952882, 0.30793370503189454)

Berdasarkan hasil uji hipotesis dengan menggunakan nilai alpha 0.05, diperoleh nilai p-value lebih besar dari alpha (0.3079 > 0.05) untuk perbandingan harga jual properti antara tahun 2017 dan 2002. Oleh karena itu, hipotesis nol (H0) diterima dan hipotesis alternatif (H1) ditolak. Kesimpulannya adalah tidak terdapat perbedaan yang signifikan dalam harga jual properti antara tahun 2017 dan 2002 di New York City.